bin/varnishd/cache/cache_vrt.c

```diff
@@ -590,6 +590,35 @@ VRT_UpperLowerStrands(VRT_CTX, VCL_STRANDS s, int up)
 	return (r);
 }
 
+// RFC7232, 3.2 without obsolete line folding:
+// ASCII VCHAR + TAB + obs-text (0x80-ff)
+static inline VCL_BOOL
+validhdr(const char *p)
+{
+	AN(p);
+	for(;*p != '\0'; p++)
+		if (vct_isctl(*p) && !vct_issp(*p))
```

Comment:
```c
/* vct.h ? */
#define vct_ishdrval(x) (!vct_isctl(x) || vct_issp(x))
```

Reply:
actually, if we touch it, we might as well have 
```c
#define vct_ishdrval(x) (((x) >= 0x20 && (x) <= 0x7e) || (x) >= 0x80)
```
which should be quite a bit more efficient (saving the table lookup).
And,  at least in my mind, it would actually be easier to understand and match to the RFC

Reply:
Or, if I get your simplification of the grammar right:

```c
#define VCT_HEADERVALUE (~VCT_CTL)
#define vct_ishdrval(x) vct_is(x, VCT_HEADERVALUE)
```

Reply:
oh sorry, I forgot to consider HTAB in my last comment, this is why I have `!vct_issp(*p)` in the PR commit.
So the simplification is
```c
#define vct_ishdrval(x) (((x) >= 0x20 && (x) <= 0x7e) || (x) >= 0x80 || (x) == 0x09)
```
`~VCT_CTL` would not work as it would match for anything but 'VCT_CTL`, e.g. `VCL_CRLF`, which we would not want.

Reply:
```c
#define vct_ishdrval(x) (((x) >= 0x20 && (x) != 0x7f) || (x) == 0x09)
```

can we compress it further? ;)

Reply:
Good point, but we can maybe find the right bit set:

```c
#define VCT_HEADERVALUE (~VCT_CTL & ~VCL_CRLF [& ~...])
```

Reply:
that would be something to miss when more labels are added to `VCT_CTL` later